### PR TITLE
feat: support generating PDF from docusaurus build directory

### PR DESCRIPTION
**Pull request by maxarndt**
Hey @KohheePeace!

At first thank you for adding me as a contributor. But unfortunately I'm not able to create/push a new branch directly to this repository. Is this by purpose or could you check the permissions?

This PR includes a feature which enables using this tool in a CD pipeline.

I added express to serve the previously built static files (normally by executing `npx docusaurus build`) and called your `generatePdf()` function afterwards.

To support this functionality I added the `from-build` command to `commander`:
```
Usage: docusaurus-pdf from-build [options] <dirPath> <firstDocPagePath> [baseUrl]

Generate PDF from a docusaurus build artifact

Options:
  -o, --output-file [name]  Specify your file name. Default is docusaurus.pdf
  -h, --help                output usage information
```

What do you think about this feature?

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -111,3 +129,27 @@ export async function generatePdf(
...
+  firstDocPath = getPathSegment(firstDocPath);
+
+  let httpServer = await app.listen();
+  let address = httpServer.address();
+  if (!address || !isAddressInfo(address)) {
+    httpServer.close();
+    throw new Error("Something went wrong spinning up the express webserver.");
+  }
+
+  app.use(baseUrl, express.static(buildDirPath));
```
Comment (maxarndt) [inline src/index.ts:151]: I have a question and hopefully someone (you?) can help me understanding:

_The function works as you can see it here, but ..._

Serving the static files failes when I move the `app.use(baseUrl, express.static(...))` command ahead of `app.listen()`. For example:

```
let app = express();
app.use(baseUrl, express.static(buildDirPath));

baseUrl = getPathSegment(baseUrl, false);
firstDocPath = getPathSegment(firstDocPath);

let httpServer = await app.listen();
...
```

I did not understand why this is happening. Normally when I work with `express` I do such `app.use()` calls always before calling the `.listen()` function.

Do you have any idea what goes wrong if I change this to the _normal_ order?

**Comment by KohheePeace** [comment]
Hi @maxarndt, I think you become a contributor because I merged your PR.
I sent invitation mail.

I will review this PR later ! Looks cool for people who wants to generate PDF during build!

File: bin/index.js [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 const chalk = require('chalk');
 const program = require('commander');
-const { generatePdf } = require('../lib');
+const { generatePdf, generatePdfFromBuildSrources } = require('../lib');
```
Comment (KohheePeace) [inline bin/index.js:5]: typo `generatePdfFromBuildSrources` => `generatePdfFromBuildSources`

File: bin/index.js [hunk 3]
```diff
@@ -22,8 +22,24 @@ program
       });
   });
 
+program
+  .command('from-build <dirPath> <firstDocPagePath> [baseUrl]')
+  .description('Generate PDF from a docusaurus build artifact')
+  .option('-o, --output-file [name]', 'Specify your file name. Default is docusaurus.pdf')
+  .action((dirPath, firstDocPagePath, baseUrl, options) => {
+    generatePdfFromBuildSrources(dirPath, firstDocPagePath, baseUrl, options.outputFile)
```
Comment (KohheePeace) [inline bin/index.js:30]: typo `generatePdfFromBuildSrources` => `generatePdfFromBuildSources`

File: src/index.ts [hunk 4]
```diff
@@ -111,3 +129,27 @@ export async function generatePdf(
   const mergedPdfBuffer = mergePdfBuffers(generatedPdfBuffers);
   fs.writeFileSync(`${filename}`, mergedPdfBuffer);
 }
+
+export async function generatePdfFromBuildSrources(
```
Comment (KohheePeace) [inline src/index.ts:133]: typo `generatePdfFromBuildSrources` => `generatePdfFromBuildSources`

**Comment by maxarndt** [comment]
Thanks for reviewing, I will fix the typos.

I‘m using Google Cloud Build as CI/CD runner. I could update the docs after the new version is released as npm package. Currently I have to :
1. `git clone` the feature branch
2. `npm install`
3. `npm run build`
4. run the program with `node`

So I think it would be much more helpful to illustrate using this program with `npx` instead with the four steps listed above. 

I already figured out that special dependencies in the execution environment are required for puppeteer. Currently I‘m using [zenika/alpine-chrome:with-puppeteer](https://hub.docker.com/r/zenika/alpine-chrome/) image to run this CD step but there are still some issues to fix. I will further analyze this on Tuesday next week. I think it is helpful to suggest a docker image which runs the PDF generation without any problems.

Do you use any CI/CD tools to give it a try?

**Comment by KohheePeace** [comment]
> So I think it would be much more helpful to illustrate using this program with npx instead with the four steps listed above.

=> okay 👍 

>  I will further analyze this on Tuesday next week

=> okay 👍 

>  I think it is helpful to suggest a docker image which runs the PDF generation without any problems.

=> Thanks! 👍 

> Do you use any CI/CD tools to give it a try?

=> No. but I will try with Netlify.

---

So, I merge this PR and publish and try during build! Thanks!

**Pull request merged**